src/extension/mongodb/src/main/java/org/geoserver/data/GeoserverWorkspaceDirecotry.java

```diff
@@ -0,0 +1,44 @@
+/* (c) 2019 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.data;
+
+import java.io.File;
+import org.geoserver.platform.resource.DataDirectoryResourceStore;
+import org.geoserver.platform.resource.Resource;
+import org.geotools.data.mongodb.data.SchemaStoreDirectoryProvider;
+import org.geotools.data.mongodb.data.StoreDirectory;
+
+/**
+ * @author ImranR
+ *     <p>Implementation injects store
+ */
+public class GeoserverWorkspaceDirecotry implements StoreDirectory {
```

Comment:
Why workspaces if the directory will stay inside GeoServer data directory?

Reply:
I would rather go for ``SchemasStoreDirectory`` instead of ``StoreDirectory``.

Reply:
agreed, will change the name